examples/jsm/libs/lottie_canvas.module.js

```diff
@@ -13925,354 +13932,14 @@ const ExpressionManager = (function () {
-      // globalData.pushExpression();
-      value = _value;
-      if (this.frameExpressionId === elem.globalData.frameId && this.propType !== 'textSelector') {
-        return value;
-      }
-      if (this.propType === 'textSelector') {
-        textIndex = this.textIndex;
-        textTotal = this.textTotal;
-        selectorValue = this.selectorValue;
-      }
-      if (!thisLayer) {
-        text = elem.layerInterface.text;
-        thisLayer = elem.layerInterface;
-        thisComp = elem.comp.compInterface;
-        toWorld = thisLayer.toWorld.bind(thisLayer);
-        fromWorld = thisLayer.fromWorld.bind(thisLayer);
-        fromComp = thisLayer.fromComp.bind(thisLayer);
-        toComp = thisLayer.toComp.bind(thisLayer);
-        mask = thisLayer.mask ? thisLayer.mask.bind(thisLayer) : null;
-        fromCompToSurface = fromComp;
-      }
-      if (!transform) {
-        transform = elem.layerInterface('ADBE Transform Group');
-        $bm_transform = transform;
-        if (transform) {
-          anchorPoint = transform.anchorPoint;
-          /* position = transform.position;
-                    rotation = transform.rotation;
-                    scale = transform.scale; */
-        }
-      }
-
-      if (elemType === 4 && !content) {
-        content = thisLayer('ADBE Root Vectors Group');
-      }
-      if (!effect) {
-        effect = thisLayer(4);
-      }
-      hasParent = !!(elem.hierarchy && elem.hierarchy.length);
-      if (hasParent && !parent) {
-        parent = elem.hierarchy[0].layerInterface;
-      }
-      time = this.comp.renderedFrame / this.comp.globalData.frameRate;
-      if (_needsRandom) {
-        seedRandom(randSeed + time);
-      }
-      if (needsVelocity) {
-        velocity = velocityAtTime(time);
-      }
-      expression_function();
-      this.frameExpressionId = elem.globalData.frameId;
-
-      // TODO: Check if it's possible to return on ShapeInterface the .v value
-      // Changed this to a ternary operation because Rollup failed compiling it correctly
-      scoped_bm_rt = scoped_bm_rt.propType === propTypes.SHAPE
-        ? scoped_bm_rt.v
-        : scoped_bm_rt;
-      return scoped_bm_rt;
-    }
-    // Bundlers will see these as dead code and unless we reference them
-    executeExpression.__preventDeadCodeRemoval = [$bm_transform, anchorPoint, time, velocity, inPoint, outPoint, width, height, name, loop_in, loop_out, smooth, toComp, fromCompToSurface, toWorld, fromWorld, mask, position, rotation, scale, thisComp, numKeys, active, wiggle, loopInDuration, loopOutDuration, comp, lookAt, easeOut, easeIn, ease, nearestKey, key, text, textIndex, textTotal, selectorValue, framesToTime, timeToFrames, sourceRectAtTime, substring, substr, posterizeTime, index, globalData];
-    return executeExpression;
+  // https://github.com/airbnb/lottie-web/issues/2828
+  // https://github.com/airbnb/lottie-web/pull/2833
+  // Bail out if we don't want expressions
+  function noOp(_value) {
```

Comment:
It's not ideal to modify library code in this way since if we want to update `lottie_canvas`, we have to apply the same fixes again. I understand the argument that lottie isn't properly maintained anymore but then we should consider to remove the dependency and the loader altogether.

`LottieLoader` isn't complex. Users should be able to call `const animation = lottie.loadAnimation( { } );` on their own and then apply lottie's canvas to a `CanvasTexture`.

Reply:
Okay, I volunteer to upstream any and all updates to/from Lottie Web for the next decade.

Does that really change anything, though? I don't think that's the problem.

Reply:
I see; maybe the class should be deprecated and the example updated to use `CanvasTexture` directly.

Reply:
@mrdoob originally added this loader. Let's see how he thinks about this issue.